src/Knp/Menu/Twig/Helper.php

```diff
@@ -142,4 +165,30 @@ private function castMenu($menu)
 
         return $menu;
     }
+
+    /**
+     * @param ItemInterface $item
+     *
+     * @return ItemInterface|null
+     */
+    private function retrieveCurrentItem(ItemInterface $item)
+    {
+        $currentItem = null;
+
+        if ($this->matcher->isCurrent($item)) {
```

Comment:
For example, in the following tree the "concrete" pages of `resources` are all on the third tier:

```
main/             
    tree/
    ...
    resources/     # route: page_list
        pages/     # route: page_list
            list   # route: page_list <-- want this one to be current
            view
            edit
        posts/
            list
            view
            edit
```

- The `resources` node could have its own route, but for now it will just show `pages.list`.
- `pages` and `pages.list` have the same route, but `pages.list` is the node which should be resolved as `current`.

So here I want the "deepest" to be used in the case where there are multiple nodes with the same route.